File: homework/fixEntity.java
import java.nio.file.Files;
import java.nio.file.Paths;
import java.util.regex.Matcher;
import java.util.regex.Pattern;

public class fixEntity {
    public static void main(String[] args) throws Exception {
        String path = "src/main/java/com/jlm/homework/entity/WrongTitleBook.java";
        String code = new String(Files.readAllBytes(Paths.get(path)));

        if (!code.contains("import com.fasterxml.jackson.annotation.JsonIgnore;")) {
            code = code.replace("import lombok.Data;", "import lombok.Data;\nimport com.fasterxml.jackson.annotation.JsonIgnore;\nimport com.fasterxml.jackson.annotation.JsonProperty;\nimport com.alibaba.cloud.commons.lang.StringUtils;");
        }

        code = code.replaceAll("@Column\(name = \"title_image\"\)\s*private String titleImage;", "@JsonIgnore\n    @Column(name = \"title_image\")\n    private String titleImage;");
        code = code.replaceAll("@Column\(name = \"source_image_url\"\)\s*private String sourceImageUrl;", "@JsonIgnore\n    @Column(name = \"source_image_url\")\n    private String sourceImageUrl;");

        String newGetters = "\n    @JsonProperty(\"titleImage\")\n" +
                "    public String getTitleImageForJson() {\n" +
                "        if (StringUtils.isNotEmpty(this.titleContext)) {\n" +
                "            return null;\n" +
                "        }\n" +
                "        return this.titleImage;\n" +
                "    }\n\n" +
                "    @JsonProperty(\"sourceImageUrl\")\n" +
                "    public String getSourceImageUrlForJson() {\n" +
                "        if (StringUtils.isNotEmpty(this.titleContext)) {\n" +
                "            return null;\n" +
                "        }\n" +
                "        return this.sourceImageUrl;\n" +
                "    }\n}\n";

        if (!code.contains("getTitleImageForJson")) {
            code = code.replaceAll("}\s*$", newGetters);
        }

        Files.write(Paths.get(path), code.getBytes());
        System.out.println("Entity modified successfully.");
    }
}

File: homework/fix_entity.py
import re

path = "src/main/java/com/jlm/homework/entity/WrongTitleBook.java"
with open(path, "r", encoding="utf-8") as f:
    code = f.read()

# Add Jackson imports if missing
if "import com.fasterxml.jackson.annotation.JsonIgnore;" not in code:
    code = code.replace("import lombok.Data;", "import lombok.Data;\nimport com.fasterxml.jackson.annotation.JsonIgnore;\nimport com.fasterxml.jackson.annotation.JsonProperty;\nimport com.alibaba.cloud.commons.lang.StringUtils;")

# Apply JsonIgnore to titleImage
code = re.sub(
    r'(\s*)@Column\(name\s*=\s*"title_image"\)(\s*)private String titleImage;',
    r'\1@JsonIgnore\n\1@Column(name = "title_image")\2private String titleImage;',
    code
)

# Apply JsonIgnore to sourceImageUrl
code = re.sub(
    r'(\s*)@Column\(name\s*=\s*"source_image_url"\)(\s*)private String sourceImageUrl;',
    r'\1@JsonIgnore\n\1@Column(name = "source_image_url")\2private String sourceImageUrl;',
    code
)

# Insert the custom getters at the end of the class before the closing brace
new_getters = """
    @JsonProperty("titleImage")
    public String getTitleImageForJson() {
        if (StringUtils.isNotEmpty(this.titleContext)) {
            return null;
        }
        return this.titleImage;
    }

    @JsonProperty("sourceImageUrl")
    public String getSourceImageUrlForJson() {
        if (StringUtils.isNotEmpty(this.titleContext)) {
            return null;
        }
        return this.sourceImageUrl;
    }
}
"""

if "getTitleImageForJson" not in code:
    code = re.sub(r'}\s*$', new_getters, code)

with open(path, "w", encoding="utf-8") as f:
    f.write(code)

print("Entity modified successfully.")
